src/Drivers/nlpSparse_ex9.hpp

```diff
@@ -0,0 +1,432 @@
+  {
+    return false;
+  }
+
+  bool eval_Jac_cons(const long long& n, 
+                     const long long& m,
+                     const double* x, 
+                     bool new_x,
+                     const int& nnzJacS, 
+                     int* iJacS, 
+                     int* jJacS, 
+                     double* MJacS)
+  {
+    assert(n==n_vars); assert(m==n_cons);
+    assert(n>=3);
+    //2*(n-1) for basecase
+
+    assert(nnzJacS == 2*(nx_-1)+S_*(2*(nx_-1)+nx_));
+
+    int nnzit{0};
+    long long conidx{0};
+
+    if(iJacS!=NULL && jJacS!=NULL) {
+      // --- constraint 1 body --->  4*x_1 + 2*x_2 == 10
+      iJacS[nnzit] = conidx;   jJacS[nnzit++] = 0;
+      iJacS[nnzit] = conidx;   jJacS[nnzit++] = 1;
+      conidx++;
+
+      // --- constraint 2 body ---> 2*x_1 + x_3
+      iJacS[nnzit] = conidx;   jJacS[nnzit++] = 0;
+      iJacS[nnzit] = conidx;   jJacS[nnzit++] = 2;
+      conidx++;
+
+      // --- constraint 3 body --->   2*x_1 + 0.5*x_i, for i>=4
+      for(auto i=3; i<nx_; i++){
+          iJacS[nnzit] = conidx;   jJacS[nnzit++] = 0;
+          iJacS[nnzit] = conidx;   jJacS[nnzit++] = i;
+          conidx++;
+      }
+      for(auto i=0;i<S_;i++) {
+        for(auto j=0;j<nx_-1;j++) {
+          iJacS[nnzit] = conidx+i*nx_+j; jJacS[nnzit] = nx_+i*nx_+j; 
+          nnzit++; 
+          iJacS[nnzit] = conidx+i*nx_+j; jJacS[nnzit] = nx_+i*nx_+j+1; 
+          nnzit++;
+        }
+        for(auto j=0;j<nx_;j++) {
+          iJacS[nnzit] = conidx+i*nx_+nx_-1; jJacS[nnzit] = nx_+i*nx_+j;
+          nnzit++;
+        }
+      } 
+      assert(nnzit == nnzJacS);
+    }
+
+    //values for sparse Jacobian if requested by the solver
+    nnzit = 0;
+    if(MJacS!=NULL) {
+      // --- constraint 1 body --->  4*x_1 + 2*x_2 == 10
+      MJacS[nnzit++] = 4;
+      MJacS[nnzit++] = 2;
+
+      // --- constraint 2 body ---> 2*x_1 + x_3
+      MJacS[nnzit++] = 2;
+      MJacS[nnzit++] = 1;
+
+      // --- constraint 3 body --->   2*x_1 + 0.5*x_4
+      for(auto i=3; i<nx_; i++){
+        MJacS[nnzit++] = 2;
+        MJacS[nnzit++] = 0.5;
+      }
+      for(auto i=0;i<S_;i++) {
+        for(auto j=0;j<nx_-1;j++) {
+          MJacS[nnzit] = -1.; 
+          nnzit++; 
+          MJacS[nnzit] = 1.; 
+          nnzit++;
+        }
+        MJacS[nnzit] = -2*(1-x[nx_+i*nx_]+xi_[i*nS_]);
+        nnzit++;
+        for(auto j=1;j<nS_;j++) {
+          MJacS[nnzit] = 2*(x[nx_+i*nx_+j]+xi_[i*nS_+j]);
+          nnzit++;
+        }
+        for(auto j=nS_;j<nx_;j++) {
+          MJacS[nnzit] = 2*x[nx_+i*nx_+j];
+          nnzit++;
+        }
+      }
+      assert(nnzit == nnzJacS);
+    }
+    return true;
+  }
+
+  /*bool eval_Hess_Lagr(const long long& n, const long long& m,
+                           const double* x, bool new_x, const double& obj_factor,
+                           const double* lambda, bool new_lambda,
+                           const int& nnzHSS, int* iHSS, int* jHSS, double* MHSS)
+  {return false;}*/
```

Comment:
Perhaps use doxygen style comments to document this interface?

Reply:
This function is not needed and has been deleted.